include/fc/container/flat.hpp

```diff
@@ -24,8 +25,8 @@ namespace fc {
          FC_ASSERT( _max_depth > 0 );
          --_max_depth;
          unsigned_int size; unpack( s, size, _max_depth );
+         FC_ASSERT( size.value <= MAX_NUM_ARRAY_ELEMENTS );
          value.clear();
-         FC_ASSERT( size.value*sizeof(T) < MAX_ARRAY_ALLOC_SIZE );
```

Comment:
I'd like to build a test to prove this, and I'd like to use an item where its size is not equal to its packed size. I haven't thought the process all the way through, but I'm thinking of a flat set of varint. Will that work? I will use small value varints for a baseline, and large value varints for the other way. I realize this does nothing for compatibility, but I believe it would help design asserts that actually check for what we want to prevent. Your thoughts please, @pmconrad.

Reply:
If you want to be realistic use object_id_type with instance values < 128.
Another example would be strings - sizeof(string) != string length.